mealie/core/security/security.py

```diff
@@ -111,16 +112,18 @@ def user_from_ldap(db: AllRepositories, username: str, password: str) -> Private
         return False
 
     if len(user_entry) > 1:
-        conn.unbind_s()
-        logger.error("[LDAP] Multiple users found with the provided user filter")
-        return False
+        logger.warning("[LDAP] Multiple users found with the provided user filter. Using the first one")
+        logger.debug(f"[LDAP] The following users were returned: {user_entry}")
 
     user_dn, user_attr = user_entry[0]
```

Comment:
Intead of just using the first result every time, I would suggest something along these lines.

```python
    if len(user_entry) > 1:
        logger.warning("[LDAP] Multiple users found with the provided user filter")

        has_id_att = False
        for entry in user_entry:
            dn, attr = entry
            if LDAP_ID_ATTRIBUTE in attr:
                has_id_att = True
                user_dn, user_attr = user_entry[0]
                logger.debug(f"[LDAP] The following users has id attribute: {entry}")
                break

        if not has_id_att:
            logger.error("[LDAP] Multiple users found with the provided attribute id")
            return False
    else:
        user_dn, user_attr = user_entry[0]
```

This should add a level of safety around MS changing the order of the responses in future versions of AD.

Reply:
I see what you're saying. Another alternative I present, we can just filter out the `user_entry` by `user_dn not None`. Because ultimately, I need the `user_dn` for the bind. So something like this could do the trick

```py
    # we only want the entries that have a dn
    user_entry = [(dn, attr) for dn, attr in user_entry if dn]

    if len(user_entry) > 1:
        logger.warning("[LDAP] Multiple users found with the provided user filter")
        logger.debug(f"[LDAP] The following entries were returned: {user_entry}")
        conn.unbind_s()
        return False;
```